CMakeLists.txt

```diff
@@ -740,13 +740,19 @@ target_include_directories (seastar
   PRIVATE
     ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
+option (BUILD_SHARED_LIBS
+  "Build seastar library as shared libraries instead of static"
+  OFF)
```

Comment:
Should we default it to ON for dev/debug?

Reply:
It probably makes sense for all users, question is what is accepted CMake practice.

Reply:
Better idea: let's change ./configure.py to set it to ON for dev/debug. Then it gets exercised during CI and this is useful for users as well.

Reply:
> Should we default it to ON for dev/debug?

no, in theory we should not set an option on behalf of user, and it's OFF by default actually. so this behavior should not be changed at least in CMake.

> Better idea: let's change ./configure.py to set it to ON for dev/debug. Then it gets exercised during CI and this is useful for users as well.

yeah, this makes more sense to me. as applications are more likely to build seastar using CMake, and it inherits parent project's setting if any.

i will create a commit to change `configure.py`.